pkg/logging/kafka/kafka_test.go

```diff
@@ -129,6 +204,113 @@ func TestUpdateKafkaInput(t *testing.T) {
 			want:      nil,
 			wantError: errors.ErrNoServiceID.Error(),
 		},
+		{
+			name: "verify SASL fields",
+			api:  mock.API{GetKafkaFn: getKafkaOK},
+			cmd:  updateCommandSASL("scram-sha-512", "user1", "12345"),
+			want: &fastly.UpdateKafkaInput{
+				Service:           "123",
+				Version:           2,
+				Name:              "log",
+				NewName:           fastly.String("log"),
+				Topic:             fastly.String("logs"),
+				Brokers:           fastly.String("127.0.0.1,127.0.0.2"),
+				RequiredACKs:      fastly.String("-1"),
+				UseTLS:            fastly.CBool(true),
+				CompressionCodec:  fastly.String("zippy"),
+				Format:            fastly.String(`%h %l %u %t "%r" %>s %b`),
+				FormatVersion:     fastly.Uint(2),
+				ResponseCondition: fastly.String("Prevent default logging"),
+				Placement:         fastly.String("none"),
+				TLSCACert:         fastly.String("-----BEGIN CERTIFICATE-----foo"),
+				TLSHostname:       fastly.String("example.com"),
+				TLSClientCert:     fastly.String("-----BEGIN CERTIFICATE-----bar"),
+				TLSClientKey:      fastly.String("-----BEGIN PRIVATE KEY-----bar"),
+				ParseLogKeyvals:   fastly.CBool(true),
+				RequestMaxBytes:   fastly.Uint(11111),
+				AuthMethod:        fastly.String("scram-sha-512"),
+				User:              fastly.String("user1"),
+				Password:          fastly.String("12345"),
+			},
+		},
+		{
+			name: "verify disabling SASL",
```

Comment:
We may be able to disregard the following because the CLI will require that credentials _always_ be supplied when SASL is enabled, but we should probably confirm that you can't enable, disable and re-renabled without supplying credentials in the API.

---

**TL;DR - we should ensure that when we disable SASL, we removed the stored SASL creds i.e., make sure these zero values make it through to the backend.**

Thanks for including this test!! It got me thinking...

I just want someone else to confirm my thoughts and that the following is true on the following:

1. enable SASL (i.e., send credentials to Fastly)
2. disable SASL 
   * This will "delete" the SASL credentials, right?
3. re-enable SASL, but don't supply credentials.

This may require us to confirm in go-fastly that sending empty string values to go-fastly will actually make it through to the Fastly backend.

---

Basically, I want to ensure that we don't have a possible security issue where if SASL was _ever_ enabled, it couldn't be disabled and then re-enabled without credentials.

Reply:
Agreed, the code to ensure that we remove the stored creds is [here](https://github.com/fastly/cli/pull/161/files#diff-5084619e19b627b734d13406b8130df2ef2018ffa290659bf4d2a1219ee51cdfR189-R193). Right now the API doesn't establish a relationship among any of the Kafka configuration fields and I reflected that in my `go-fastly` changes. I think ideally we'd implement this logic once in the API versus in each consumer, but for now I wanted to deal with it here until I worked out the best way to deal with it more generally.